### PR TITLE
Introduce finer version control for codestarts and use it 

### PR DESCRIPTION
**Pull request by ia3andy**
This fixes project generation incompatibility issues introduced by https://github.com/quarkusio/quarkus/pull/36826

### CODE REVIEW

**Comment by ia3andy** [comment]
If we see this version `if` usage is growing up under control, we will re-evaluate the strategy.

**Review by aloubyansky — APPROVED** [review]

**Comment by ia3andy** [comment]
I switched to Jackson Version instead of the Java Runtime one, because the Java one is not working with 1.0

**Review by aloubyansky — DISMISSED** [review]
I'll check whether we can use a better version comparison approach

**Comment by ia3andy** [comment]
> I'll check whether we can use a better version comparison approach

@aloubyansky we can always change later, this needs no dependency.

I know Maven has a better one but it wasn't in their..

File: independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/reader/QuteCodestartFileReader.java [hunk 1]
```diff
@@ -6,9 +6,13 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.CompletionStage;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import org.apache.commons.io.FilenameUtils;
 
+import com.fasterxml.jackson.core.Version;
```
Comment (gastaldi) [inline independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/reader/QuteCodestartFileReader.java:14]: I'm not a big fan of depending on Jackson here. `io.smallrye.common:smallrye-common-version` may be a better substitute here

**Review by maxandersen — COMMENTED** [review]
How does this affect backwards compatibility if any? Old clis will continue to work?

File: independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/reader/QuteCodestartFileReader.java (hunk 1 above)
Comment (ia3andy) [reply independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/reader/QuteCodestartFileReader.java]: Fixed, used Maven one

**Comment by ia3andy** [comment]
@aloubyansky could you merge this?

File: independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/reader/QuteCodestartFileReader.java [hunk 2]
```diff
@@ -6,30 +6,24 @@
...
-import io.quarkus.qute.Variant;
+import io.quarkus.qute.*;
 
 final class QuteCodestartFileReader implements CodestartFileReader {
 
     private static final String TPL_QUTE_FLAG = ".tpl.qute";
     private static final String ENTRY_QUTE_FLAG = ".entry.qute";
     public static final String INCLUDE_QUTE_FLAG = ".include.qute";
     public static final String SKIP_TAG = "<SKIP>";
+    private static final Pattern VERSION_PATTERN = Pattern.compile("^(\\d+)(\\.(\\d+))?(\\.(\\d+))?.*");
```
Comment (aloubyansky) [inline independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/reader/QuteCodestartFileReader.java:26]: I guess we don't need this constant anymore and `splitVersion(...)`?

Comment (ia3andy) [reply independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/reader/QuteCodestartFileReader.java:26]: right, duuuuuuhhhh another CI run 😬

**Review by aloubyansky — APPROVED** [review]

**Pull request merged**